jetstream/resource_jetstream_stream.go

```diff
@@ -343,9 +343,14 @@ func resourceStreamRead(d *schema.ResourceData, m any) error {
 		return fmt.Errorf("could not load stream %q: %s", name, err)
 	}
 
-	maxAge := str.MaxAge().Seconds()
-	if maxAge == 0 {
-		maxAge = -1
+	// str.Compression returns an api.Compression type. We cannot turn this into a string with .String()
+	// without breaking our resource api, since .String() will return "s2 compression" instead of "s2".
+	compression := "unknown"
```

Comment:
can also probably do `json.Marshal(str.Compression())` which will work forever even as we add more compression types in future, this will also fail on the unknown type outcome which would be the correct behaviour - we dont want people to set invalid types here (though that shouldn't happen as the schema calls `validateCompressionTypeString()`

Reply:
Might be worth updating `validateCompressionTypeString()` to also use the marshaller to keep things working long term without maintenance